README.md

```diff
@@ -49,6 +49,15 @@ To run the pre-commit hook test:
 tox -e py37-hook
 ```
 
+### Required changes for a new plugin
+This demonstration is setup with a plugin named `plugin` there are a number of locations to change. At a top level for a plugin
+`foo` at least the following changes are required
+- Global find and replace `mdformat_plugin` to `mdformat_foo` including folder names.
+- Global find and replace `mdformat-plugin` to `mdformat-foo` including folder names.
+- `tests/test_fixtures.py`: `output = render_text(text, extensions={"plugin"})` becomes `output = render_text(text, extensions={"foo"})`
+- `pyproject.toml` in addition to the global find and replace: `plugin = "mdformat_plugin"` becomes `foo = "mdformat_foo"`
+
+Do not forget to update authorship / maintainers in `pyproject.toml` as well.
```

Comment:
I think I'd move this section all the way to the top, as the first second level (`##`) heading. This seems like the most important section for someone who's creating their own plugin.

What does @chrisjsewell think?

Reply:
Yep sounds reasonable.
Note, the other way to do templating is to create a https://github.com/cookiecutter/cookiecutter package. This would automate required changes, but the downside (and why I opted not to) is that it is harder to develop and test/maintain